### PR TITLE
feat: make remote client configurable

### PR DESCRIPTION
**Pull request by MachaelLee**
## Related Issues
Closes #

## Detailed Changes

* add compression config in remote client config 
* add remote client config in server config

### CODE REVIEW

File: table_engine/src/remote/model.rs [hunk 1]
```diff
@@ -159,15 +157,16 @@ impl TryFrom<ceresdbproto::remote_engine::WriteBatchRequest> for WriteBatchReque
     }
 }
 
-impl TryFrom<WriteBatchRequest> for ceresdbproto::remote_engine::WriteBatchRequest {
-    type Error = Error;
-
-    fn try_from(batch_request: WriteBatchRequest) -> std::result::Result<Self, Self::Error> {
+impl WriteBatchRequest {
+    pub fn batch_convert_to_pb(
```
Comment (Rachelint) [inline table_engine/src/remote/model.rs:161]: convert_write_batch_to_pb?

Comment (MachaelLee) [reply table_engine/src/remote/model.rs]: Ok, fixed.

File: components/arrow_ext/src/ipc.rs [hunk 2]
```diff
@@ -48,13 +51,22 @@ pub struct RecordBatchesEncoder {
...
     /// The minimum length of the payload to be compressed.
     pub compress_min_length: usize,
     pub method: CompressionMethod,
 }
 
+impl Default for CompressOptions {
+    fn default() -> Self {
+        Self {
+            compress_min_length: DEFAULT_COMPRESS_MIN_LENGTH,
+            method: CompressionMethod::None,
```
Comment (jiacai2050) [inline components/arrow_ext/src/ipc.rs:65]: Those value default to zstd before, why change to none?

Comment (MachaelLee) [reply components/arrow_ext/src/ipc.rs:65]: In my opinion, in most scenarios, network traffic is not the limiting factor for a system, so there is no need to default to using zstd compression.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**